### PR TITLE
feat: generate random name + avatar

### PR DESCRIPTION
**Pull request by RyanMatthewJacobs**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Database now stores usernames and user images in the users table. Username is randomly generated and checked for collisions with other names in the party. User image is a url to an image hosted by Dicebear.

- **What is the current behavior?** (You can also link to an open issue here)

- **What is the new behavior (if this is a feature change)?**

- **Other information**:

### CODE REVIEW

File: src/lib/services/party.ts [hunk 1]
```diff
@@ -41,4 +41,26 @@ export default class Party {
...
+  static async getPartyWithUsers({ partyHash }: GetParams) {
+    return dbClient.parties.findFirst({
+      where: {
+        hash: partyHash,
+      },
+      include: {
+        users: true,
+      },
+    });
+  }
```
Comment (rkrishn7) [inline src/lib/services/party.ts]:
```suggestion
  static async searchUsers({ partyHash }: SearchUsersParams) {
    return dbClient.parties.findFirst({
      where: {
        hash: partyHash,
      },
      select: {
        users: true,
      },
    });
  }
```

This should exclusively *select* users which should eliminate the need for the below method

File: src/lib/services/party.ts [hunk 2]
```diff
@@ -41,4 +41,26 @@ export default class Party {
...
+  static async getCurrentUsernames({ partyHash }: GetParams): Promise<Set<string>> {
+    const party = await Party.getPartyWithUsers({
+      partyHash: partyHash,
+    });
+    const usernames: string[] = [];
+    for (const user of party.users) {
+      usernames.push(user.name);
+    }
+    return new Set(usernames);
+  }
```
Comment (rkrishn7) [inline src/lib/services/party.ts]: We can remove

File: src/lib/services/users.ts [hunk 3]
```diff
@@ -63,9 +65,43 @@ export default class Users {
...
+     * w : width of icon
+     * h : height of icon
+     * colorLevel : brightness of color
+     * 
+     * EX: ?r=4&m=2&b=%23b99d9d&w=1&h=1&colorLevel=800
+     */
+    const options = ""; // I'm leaving this blank for now, but we'll want to add options so the image fits nicely in the chat window
+    const baseUrl = "https://avatars.dicebear.com/api/identicon/";
+    return baseUrl + encodeURIComponent(userName) + ".svg" + options;
+  }
```
Comment (rkrishn7) [inline src/lib/services/users.ts]: Let's use `stringifyUrl` and pass in query params as an object. Also, we can leave a link to the options.

File: src/lib/services/users.ts [hunk 4]
```diff
@@ -63,9 +65,43 @@ export default class Users {
...
+      style: 'capital',
+      length: 2,
+    };
+    let randName: string;
+    do {
+      randName = uniqueNamesGenerator(nameConfig);
+    } while (set.has(randName));
+    console.log("New name:", randName);
+    return randName;
+  }
```
Comment (rkrishn7) [inline src/lib/services/users.ts]: With the updated method, the predicate in the while loop becomes, assuming an import from `lodash`:

```ts
find(users, u => u.name === randName)
```

**Review by rkrishn7 — COMMENTED** [review]
Looks good! It's basically there, just mostly some style suggestions. I pulled it down to test so I'm gonna push a commit with my updated suggestions.

**Comment by RyanMatthewJacobs** [comment]
> Looks good! It's basically there, just mostly some style suggestions. I pulled it down to test so I'm gonna push a commit with my updated suggestions.

Thanks for the review @rkrishn7! I should definitely look into lodash.

**Pull request merged**